### DISCUSSION TITLE
Building PetSc & pytest

### DISCUSSION THREAD

**Discussion by Jcini** [firedrakeproject/firedrake#2753] [Firedrake support] [answered]
Hi, I'm using homebrew Python 3.10.9 on M1 macOS 12.6.

I have the following error when install PetSc:

> Traceback (most recent call last):
  File "/Users/jcini/firedrake-install", line 1854, in <module>
    install("petsc/")
  File "/Users/jcini/firedrake-install", line 1022, in install
    build_and_install_petsc()
  File "/Users/jcini/firedrake-install", line 1161, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "/Users/jcini/firedrake-install", line 655, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 421, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/jcini/firedrake/bin/python', './configure', 'PETSC_DIR=/Users/jcini/firedrake/src/petsc', 'PETSC_ARCH=default', '--download-mpich', '--LDFLAGS=-Wl,-rpath,/opt/homebrew/opt/openblas/lib -L/opt/homebrew/opt/openblas/lib', '--download-ml', '--download-scalapack', '--download-metis', '--download-ptscotch', '--with-fortran-bindings=0', '--with-c2html=0', '--download-superlu_dist', '--download-mpich-configure-arguments=--disable-opencl', '--download-pastix', '--CFLAGS=-I/opt/homebrew/opt/openblas/include -Wno-implicit-function-declaration', '--with-shared-libraries=1', '--download-hypre', '--download-hwloc', '--download-pnetcdf', '--download-netcdf', '--download-mumps', '--with-zlib', '--download-bison', '--download-eigen=/Users/jcini/firedrake/src/eigen-3.3.3.tgz ', '--download-chaco', '--download-hdf5', '--with-x=0', '--with-blaslapack-dir=/opt/homebrew/opt/openblas', '--download-hwloc-configure-arguments=--disable-opencl', '--download-suitesparse', '--with-debugging=0']' returned non-zero exit status 1.

I have the following log file: 
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/10611795/firedrake-install.log)

In this case, when I tried to run 

> pytest tests/regression/ -k "poisson_strong or stokes_mini or dg_advection"

I got

> command not found: pytest

I saw discussion #2605 , and then I uninstall and reinstall Xcode Commend Line Tool. But I still get same error. 

I wonder what should I do now. Thank you so much!

**Comment by connorjward** [comment, marked as answer]
Please share `configure.log` from inside `firedrake/src/petsc`. I also notice that `brew doctor` raises a number of warnings. I don't know if these are causing your installation to fail but they may be worth resolving.

**Reply by Jcini** [reply]
Hi, thank you. I fixed the brew warnings and then try to install firedrake again, but it still not working. I will attach the new log files here. Thanks!

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/10679806/firedrake-install.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/10679810/configure.log)

**Reply by connorjward** [reply]
I think you are having the same issue as: https://github.com/firedrakeproject/firedrake/issues/2689. Please do as it states and set `MACOSX_DEPOLYMENT_TARGET` inside the `firedrake-install` script.

**Reply by Jcini** [reply]
Thank you! It works!

**Comment by JoshArntx** [comment]
It seems that the installation of PetSc failed due to an error during the configuration step. The error message suggests that some required dependencies are missing or not properly configured. One possibility is that the installation of PetSc requires some additional system packages or libraries that are not installed on your machine.

To solve the issue, you can try the following steps:

Check the installation logs of PetSc for more detailed error messages. The logs should contain more information about the specific error that occurred during the configuration step.

Check if any required system packages or libraries are missing by looking at the installation instructions for PetSc. You may need to install additional dependencies manually, or using a package manager like Homebrew.

Try running the installation again, making sure to follow the installation instructions carefully and ensuring that all required dependencies are installed and properly configured.

Regarding the second issue, it seems that the pytest command is not found on your system. This may be due to the fact that the pytest package is not installed or not in your system's PATH. To install pytest, you can use the following command:

pip install pytest

After installing pytest, make sure to add its location to your system's PATH environment variable so that it can be found by other commands. You can do this by adding the following line to your shell profile (e.g. .bashrc, .zshrc, etc.):

export PATH="$HOME/Library/Python/3.10/bin:$PATH"

Replace 3.10 with the version of Python you are using if necessary. After making this change, restart your shell or run source ~/.bashrc (or equivalent) to apply the changes.

I hope this helps!
